### DISCUSSION TITLE
Nitro as a microservice without HTTP API

### DISCUSSION THREAD

**Discussion by MickL** [nitrojs/nitro#2752] [Ideas]
For me personally there is one thing missing for Nitro to be a "complete package" for all my web projects:

Use Nitro not as a HTTP API, but as a microservice that listens to incoming requests from other sources like NATS, AWS SNS/SQS, Redis Pub/Sub, or similar. Also a hybrid solution is thinkable: One Nitro app that provides a public HTTP api but also listen for microservice events.

Probably this is not what Nitro wants to be, but what I like about Nitro (and Nuxt) is that it takes everything off my hands:

- Create a new project with the CLI that just works
- Dev environment with file watching
- Bundling, HMR, Tree Shaking
- Bundling for basically every environment (Node, Bun, Deno, Cloudflare, Vercel, ...)
- Best practices
- Integration of UnJS ecosystem as well as no restrictions in using other frameworks (e.g. Zod, Drizzle)

It would be amazing to have all this for a microservice as well!

**Comment by chadxz** [comment]
What do you think Nitro is missing to be viable for what you are describing?

**Comment by laplaceliu** [comment]
In fact, you can do anything in the plugin, including NATS, AWS SNS/SQS, Redis Pub/Sub as you mentioned. You can use the plugin as an entry point to start your microservices here.

- https://nitro.build/guide/plugins
- https://nuxt.com/docs/guide/directory-structure/plugins
